src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    readonly color: Color,
+    readonly a: number,
+    readonly b: number,
+    readonly c: number,
+  ) {
+    if (a < 1 || b < 1 || c < 1) {
+      throw new Error('Bad input data!');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
try to avoid such long conditions inside if, you can create variable for it, for example: `isInvalidInput = a<1 || b<1 || c<1` and i think that we should have ability to get number with decimals, therefore `a<=0`, `b<=0`, `c<=0`
`a + b <= c || a + c <= b || b + c <= a` -> here you can get sum of all and subtract a max value, but optional for you, and put condition into variable